packages/tempus/src/Tempus_TimeStepControlStrategy.hpp

```diff
@@ -19,52 +19,60 @@ namespace Tempus {
 
 template<class Scalar> class TimeStepControl;
 
-/** \brief StepControlStrategy class for TimeStepControl
+/** \brief TimeStepControlStrategy class for TimeStepControl
  *
- *  This strategy is the default base class, which provides a no-op strategy.
+ *  This is the base class for TimeStepControlStrategies.
+ *  The primary function required from derived classes is setNextTimeStep(),
+ *  which will
+ *   - determine the next step from information in the TimeStepControl
+ *     and SolutionHistory (i.e., SolutionStates)
+ *   - set the next time step on the workingState in the SolutionHistory
+ *  If a valid timestep can not be determined the Status is set to FAILED.
  */
 template<class Scalar>
 class TimeStepControlStrategy
   : virtual public Teuchos::Describable,
-    virtual public Teuchos::ParameterListAcceptor
+    virtual public Teuchos::VerboseObject<Tempus::TimeStepControlStrategy<Scalar> >
 {
 public:
 
-  /// Constructor
-  TimeStepControlStrategy(){}
+  virtual std::string getStepType() const { return stepType_; }
 
-  /// Destructor
-  virtual ~TimeStepControlStrategy(){}
-
-  /// Determine the time step size.
+#ifndef TEMPUS_HIDE_DEPRECATED_CODE
+  /// Deprecated get the time step size.
   virtual void getNextTimeStep(
-    const TimeStepControl<Scalar> /* tsc */,
+    const TimeStepControl<Scalar> & tsc,
+    Teuchos::RCP<SolutionHistory<Scalar> > sh,
+    Status & /* integratorStatus */)
+  {
+    this->setNextTimeStep(tsc, sh, integratorStatus);
+  };
```

Comment:
Compilation error here, uncomment the `integratorStatus`

Reply:
Yeah, I was compiling with Tempus_HIDE_DEPRECATED_CODE=ON, so it missed this.  I have fixed it.  The bigger problem is with EMPIRE and the embedded strategy in EMPIRE-Fluid.  My changes have serious broke it.  :(  So I need to rework the TimeStepControlStrategyIntergalController, and probably put back in the ParameterList along with these changes so I can switch EIMPIRE-Fluid over to the new implementation, and then remove the ParameterList in another PR.